nixos/modules/services/cluster/k3s/default.nix

```diff
@@ -33,15 +33,40 @@ in
 
     serverAddr = mkOption {
       type = types.str;
-      description = lib.mdDoc "The k3s server to connect to. This option only makes sense for an agent.";
+      description = lib.mdDoc ''
+        The k3s server to connect to.
+
+        Servers and agents need to communicate each other. Read
+        [the networking docs](https://rancher.com/docs/k3s/latest/en/installation/installation-requirements/#networking)
+        to know how to configure the firewall.
+      '';
       example = "https://10.0.0.10:6443";
       default = "";
     };
 
+    clusterInit = mkOption {
+      type = types.bool;
+      default = cfg.role == "server";
+      defaultText = literalExpression ''cfg.role == "server"'';
+      example = false;
+      description = lib.mdDoc ''
+        Initialize a cluster if k3s is started for the first time.
+
+        If the cluster is already initialized, this option has no effect.
```

Comment:
From [the docs](https://rancher.com/docs/k3s/latest/en/installation/ha-embedded/#existing-clusters), it sounds like this does have an effect on existing clusters.

As it says:

> If you have an existing cluster using the default embedded SQLite database, you can convert it to etcd by simply restarting your K3s server with the --cluster-init flag

Since I think sqlite is / has-been the default, that means many existing clusters will be using it, and adding this default will migrate them forward to etcd.

The upstream default is still sqlite, and you have to opt-in to etcd, so I feel like that's the better default for us.

Reply:
I can expose the `--datastore-endpoint` option. Then assert that it is `etcd` if we're enabling cluster-init.

Finally I can change the defaults to be sqlite without cluster-init, to stick closer to upstream.

Does that seem a good solution?

Reply:
Reading https://rancher.com/docs/k3s/latest/en/installation/datastore/#external-datastore-configuration-parameters I can see that `--datastore-endpoint` shall be kept empty for any embedded DB usage.

It seems like k3s is smart enough to use embedded sqlite by default and to migrate to embedded etcd automatically when the server gets `--cluster-init` or when another server joins the 1st one.

So I'll just keep this option `false` by default and improve the docs to let users understand that this flag enables embedded etcd mode.